lib/services/database_service.dart

```diff
@@ -12,15 +12,24 @@ import 'package:adventures_in_chat_app/extensions/extensions.dart';
 class DatabaseService {
   String currentUserId;
   final Database _database;
-  final _controller = StreamController<List<MessagesListItem>>();
+  final _controllers = <String, StreamController<List<MessagesListItem>>>{};
 
   DatabaseService({Database database})
       : _database = database ?? FirestoreDatabase();
 
   Stream<List<MessagesListItem>> getMessagesStream(String conversationId) {
+    // fresh controller per unique conversation.
+    StreamController<List<MessagesListItem>> _controller;
+    if (_controllers.containsKey(conversationId)) {
+      _controller = _controllers[conversationId];
+    } else {
+      _controller = StreamController<List<MessagesListItem>>();
+    }
+
     _database.getMessagesStream(conversationId).listen((messagesList) {
-      var latest = messagesList.first.timestamp;
-      final newList = <MessagesListItem>[SectionDate(latest)];
+      var latest = DateTime.fromMicrosecondsSinceEpoch(0); // init to minimum
```

Comment:
Correct me if I'm wrong - this change has nothing to do with bug right?  If you'd like to make the change can it go up as it's own issue/PR so it can be discussed?

Reply:
I’ll create the Additional issue.

Reply:
I added #208 to track this issue / solution seperately.